addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,275 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import json
+from collections import defaultdict
+
+from odoo import Command, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    # ------------------
+    # Fields declaration
+    # ------------------
+
+    display_withholding = fields.Boolean(compute='_compute_display_withholding')
+    withhold_tax = fields.Boolean(
+        string='Withhold Tax Amounts',
+        compute='_compute_withhold_tax',
+        readonly=False,
+        store=True,
+    )
+    withholding_line_ids = fields.One2many(
+        string="Withholding Lines",
+        comodel_name='account.payment.register.withholding.line',
+        inverse_name='payment_register_id',
+        compute='_compute_from_lines',
+        store=True,
+        readonly=False,
+    )
+    hide_withholding_number_col = fields.Boolean(compute='_compute_hide_withholding_number_col')
+    withholding_net_amount = fields.Monetary(
+        string='Net Amount',
+        help="Net amount after deducting the withholding lines",
+        compute='_compute_withholding_net_amount',
+    )
+    # We need to define the outstanding account of the payment in order for it to have the proper journal entry.
+    # To that end, we'll have this field required if we have a withholding tax impacting the payment, and we don't have a payment account set on the payment method.
+    default_account_id = fields.Many2one(
+        related='journal_id.default_account_id'
+    )
+    outstanding_account_id = fields.Many2one(
+        comodel_name='account.account',
+        string="Outstanding Account",
+        copy=False,
+        domain="[('deprecated', '=', False), '|', ('account_type', 'in', ('asset_current', 'liability_current')), ('id', '=', default_account_id)]",
+        check_company=True,
+        compute="_compute_outstanding_account_id",
+        store=True,
+        readonly=False,
+    )
+    payment_account_id = fields.Many2one(related="payment_method_line_id.payment_account_id")
+    payment_move_amount_total = fields.Monetary(
+        compute="_compute_payment_move_amount_total",
+    )
+    hide_tax_base_account = fields.Boolean(compute='_compute_hide_tax_base_account')
+
+    # --------------------------------
+    # Compute, inverse, search methods
+    # --------------------------------d
+
+    @api.depends('withholding_line_ids.amount', 'amount')
+    def _compute_withholding_net_amount(self):
+        for wizard in self:
+            wizard.withholding_net_amount = wizard.amount - sum(wizard.withholding_line_ids.mapped('amount'))
```

Comment:
You are on the payment register wizard. You can't do that without considering the `can_edit_wizard` boolean. You have to make it working on multiple invoices too.

Reply:
> You have to make it working on multiple invoices too.
We decided only to support registering withholding taxes on single invoice payments for now.
If you start mixing multiple records, it could quickly become a mess with the table in the view.